### PR TITLE
Revert input_columns change

### PR DESCRIPTION
**Pull request by lhoestq**
Revert https://github.com/huggingface/datasets/pull/4971

Fix https://github.com/huggingface/datasets/issues/5005

### CODE REVIEW

File: tests/test_arrow_dataset.py [hunk 1]
```diff
@@ -1362,14 +1362,12 @@ def test_map_input_columns(self, in_memory):
         with tempfile.TemporaryDirectory() as tmp_dir:
             with self._create_dummy_dataset(in_memory, tmp_dir, multiple_columns=True) as dset:
                 with dset.map(lambda col_1: {"label": col_1 % 2}, input_columns="col_1") as mapped_dset:
-                    self.assertEqual(mapped_dset[0].keys(), {"col_1", "col_2", "col_3", "label"})
+                    self.assertEqual(mapped_dset[0].keys(), {"col_1", "label"})
```
Comment (albertvillanova) [inline tests/test_arrow_dataset.py:1365]: Why don't you remove completely the test `test_map_input_columns`, introduced in #4971?

**Review by albertvillanova — COMMENTED** [review]
Just a question below.

File: tests/test_arrow_dataset.py (hunk 1 above)
Comment (lhoestq) [reply tests/test_arrow_dataset.py:1365]: Unfortunately input_columns wasn't tested before this PR, so I think it's nice to keep this test, but with the correct behavior (i.e. don't keep all the columns)

**Review by albertvillanova — APPROVED** [review]
Seems good. Thanks.

**Comment by lhoestq** [comment]
Merging this one and I'll check if it fixes the `transformers` CI before doing a patch release

**Pull request merged**